### ISSUE TITLE
boxr::box_auth() require authorization code

### ISSUE THREAD

**Issue by jtraurig** [r-box/boxr#191]
https://r-box.github.io/boxr/articles/boxr-app-interactive.html

I have used the above instructions to attempt to authorize box use on an Rstudio server. While this works for our local serves this process does not work for RStudio servers running on AWS Virtual machines. Is there some way around a need to have to type in an authorization code with using box_auth() on Rstudio server

![Capture](https://user-images.githubusercontent.com/55104463/95128220-1a391680-0727-11eb-972e-a40552f559ca.JPG)

![Capture2](https://user-images.githubusercontent.com/55104463/95128229-1dcc9d80-0727-11eb-81c6-2aaaf856a2e4.JPG)

**Comment by nathancday** [comment]
Have you seen the section about using on a remote RStudio instance?
https://r-box.github.io/boxr/articles/boxr-app-interactive.html#transfer

There is also JWT based method that doesn't require the browser confirmation, but does require a Box service account. 
https://r-box.github.io/boxr/articles/boxr-app-service.html

**Comment by jtraurig** [comment]
Yes. I have seen the section about using on remote server. I have followed
the instructions. Unfortunately uploading the .box-oauth file and adding
credentials to .Renviron does not work.

**Comment by jtraurig** [comment]
Update. I think I've made some progress. The "httr_oob_default" must = FALSE. That gets me passed the part where R is requesting and authorization key but now it's asking me to input my box credentials again even though I already did that on my local machine.  I have followed the exact same steps on a local Rstudio server version and everything works fine. This is only an issue for a AWS virtual machine running Rstudio server. Any ideas?

**Comment by ijlyttle** [comment]
This might be a long shot, and you would have already read about this, but are the versions of httr identical on both the server and on the local computer where you generate the boxr token?

**Comment by jtraurig** [comment]
Yeah. I did catch that in the instructions. They are the same version 1.4.2 unfortunately. If it helps, I've narrowed down in code where the prompt is happening. Screen shots below:

This is in box_auth
![image](https://user-images.githubusercontent.com/55104463/95204298-ec4ce400-07b1-11eb-9708-a69967eec8a8.png)

And then I step into httr::oauth2.0_token() and the prompt happens at the Token2.0$new() call. Not that familiar with what is going on here. Any ideas what might be causing the credential prompt to open in another window.

Note, if I haven't said this before. I have followed the exact same steps using a local Rstudio Server version and have successfully authenticated, which is frustrating since it works there but not virtual machine.

**Comment by nathancday** [comment]
Using `box_auth_service()` is the proper way to handle this type of server-server authentication. I'd recommend going down that road.

Interactive OAuth was never meant for use on remote machines, but the copying of the token was an easy hack/work-around so we did it 🤷

**Comment by jtraurig** [comment]
No problem. Thanks for the help. I did eventually figure out a solution. The issue was related to the .box-oauth file which is just a rds file (which I'm sure you know). The .rds file has a function to generate a hash which should match the name of the list in the .rds file; however, the hash created on my windows machine was different than the hash created on the server. Because of this, the .boxr-oauth file was actually being ignored on the server. The hash generated on the server did not match the list name of the .rds file. I was able to work around this with a quick hack.

That said, the service route does appear to be a better path. Saving the json token on multiple machines is much easier than figuring the system differences for each machine.
